mintscript/src/script/mod.rs

```diff
@@ -121,11 +121,24 @@ impl Threshold {
     }
 }
 
+#[derive(Clone, PartialEq, Eq, Debug)]
+pub enum HashType {
```

Comment:
I'd say `HashType` is no longer an accurate name since it now actually includes the hash itself. Not that I have a great suggestion.

Reply:
Idk, maybe just `Hash`?

Reply:
> Idk, maybe just `Hash`?

I would say why not but it conflicts with the `Hash` trait from standard library so not quite sure. How about something like `HashChallenge`? I don't know if this is good terminology since the work challenge is used in cryptography but I don't know how exactly so not sure if this is vaguely appropriate. @TheQuantumPhysicist

Reply:
Maybe Hashed, just to avoid the collision with std? Challenge seem to be OK as well. I'll have to think about it. Pick what you find appropriate now.

Reply:
renamed to `HashChallenge `